uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/PublicHolidayLeaveRequestCreationTest.php

```diff
@@ -101,4 +102,17 @@ public function testCanCreatePublicHolidayLeaveRequestsForAnAbsenceType() {
 
     $this->assertEquals(-2, LeaveBalanceChange::getLeaveRequestBalanceForEntitlement($periodEntitlement));
   }
+
+  /**
+   * @expectedException InvalidArgumentException
+   * @expectedExceptionMessage It's not possible to create Public Holidays for Absence Types where 'Must take public holiday as leave' is false
```

Comment:
mmm I see ... but chaning the expection message will ... I really don't like the idea of relying on exepction message I feel they are just wrong and codes are more robust ... but I am not gonna push on this since it will require a lot more work .